core/src/main/java/io/smallrye/openapi/runtime/io/schema/SchemaFactory.java

```diff
@@ -340,7 +340,7 @@ public static Schema includeTypeSchema(AnnotationScannerContext context, Schema
         if (schema.getType() != null && schema.getType().contains(Schema.SchemaType.ARRAY) && implSchema != null) {
             // If the @Schema annotation indicates an array type, then use the Schema
             // generated from the implementation Class as the "items" for the array.
-            schema.setItems(implSchema);
+            schema.setItems(lookupRef(context, type, implSchema));
```

Comment:
This change seems contradictory to line 337, where we call `implSchema = readClassSchema(context, type, false);` and `false` here indicates that we don't want a reference.

Similarly, I think passing `false` there will prevent `readClassSchema` from adding the schema to the registry in the first place. Where do we expect the item schema to be added to the registry?

If we're expecting the item schema to be added to the registry some time before this, is it guaranteed that the item schema is processed before this schema?

I note that `readClassSchema` also calls `setItems` - does that also need to look up references?

Reply:
The flag passed to allow schema registration or not will only enable/disable returning a reference schema. The call to `OpenApiDataObjectScanner#process` may still register the schema, but that method will not return the reference itself. It is up to the SchemaRegistry to do that by possibly calling `SchemaRegistry#schemaRegistration`.

> If we're expecting the item schema to be added to the registry some time before this, is it guaranteed that the item schema is processed before this schema?

Yes, that should occur within `OpenApiDataObjectScanner` for `ClassType`s.

> I note that readClassSchema also calls setItems - does that also need to look up references?

The last commit also updates the two other `setItems` locations in `SchemaFactory` and adjusts the tests accordingly.